PlayList.java

```diff
@@ -71,34 +92,60 @@ public int indexOf(String title) {
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
     public boolean add(int i, Track track) {
-        //// replace the following statement with your code
+        if (this.size != this.maxSize && i >= 0 && i <= this.size){
+            for (int j = this.size; j > i ; j--){
+                this.tracks[j] = this.tracks[j-1];
+            }
+            this.tracks[i]= track;
+            this.size++;
+            return true;
+        }
         return false;
     }
      
     /** Removes the track in the given index from this list.
      *  If the list is empty, or the given index is negative or too big for this list, 
      *  does nothing and returns -1. */
     public void remove(int i) {
-        //// replace this comment with your code
+        if (this.size > 0 && i >= 0 && i <= this.maxSize){
+            for (int j = i; j < this.size-1 ; j++){
+                this.tracks[j] = this.tracks[j+1];
+            }
+            this.tracks[this.size - 1]= null;
+            this.size--;
+        }
     }
 
+
     /** Removes the first track that has the given title from this list.
      *  If such a track is not found, or the list is empty, or the given index
      *  is negative or too big for this list, does nothing. */
     public void remove(String title) {
-        //// replace this comment with your code
+        if (this.size>0){
+            int index = indexOf(title);
+            if (index >= 0 && index < this.tracks.length){
+                remove(index);
+                this.size--;
```

Comment:
Size decreasing happening in `remove(index)`, this lead wrong size when using `remove(title)` (size will be reduced by 2)
```suggestion
```